ray-operator/controllers/ray/common/pod.go

```diff
@@ -875,51 +874,3 @@ func findMemoryReqOrLimit(container corev1.Container) (res *resource.Quantity) {
 	}
 	return nil
 }
-
-// ValidateHeadRayStartParams will validate the head node's RayStartParams.
```

Comment:
Agree that we should let Ray do the validation. One question is that if Ray validation fails, is user able to see the error message and which is wrong?

Reply:
If the error message is propagated to the STDOUT or STDERR of the "ray start" process, users will be able to see it. For example, I set `rayStartParams` as follows, and then I can see the error message in the head Pod's log.

```yaml
rayStartParams:
  memory: "abc"
```

<img width="1438" alt="Screen Shot 2024-04-05 at 4 33 23 PM" src="https://github.com/ray-project/kuberay/assets/20109646/2b517b41-27cd-40a8-935c-569f79b621ae">

Reply:
I agree that validating the start params from KubeRay is not necessary. But if Ray head is crash looping due to bad params, we should also surface cluster state as either `unhealthy` or `failed` right? From @Yicheng-Lu-llll's test in https://github.com/ray-project/kuberay/pull/2068#issuecomment-2040768976 the state is not updated at all.

Reply:
> But if Ray head is crash looping due to bad params, we should also surface cluster state as either unhealthy or failed right?

Yes, we should add a field in [HeadInfo](https://github.com/ray-project/kuberay/blob/d3b7d96ff56d7a45c9afc5eb1c78cd7ae8c8bb1a/ray-operator/apis/ray/v1/raycluster_types.go#L147) to indicate whether the head Pod is ready or not as I mentioned in https://github.com/ray-project/kuberay/pull/1930#pullrequestreview-1978124073.